packages/taquito/src/taquito.ts

```diff
@@ -308,7 +308,7 @@ export class TezosToolkit {
    */
   setReadProvider(readProvider?: SetProviderOptions['readProvider']) {
     const readP =
-      typeof readProvider === 'undefined' ? this.getFactory(RpcReadAdapter)() : readProvider;
+      typeof readProvider === 'undefined' ? new RpcReadAdapter(this._context.rpc) : readProvider;
```

Comment:
again unrelated to the change 
can the type of readProvider be another falsy value other than undefined? if not it could be simplified with just 
`readProvider ? new RpcReadAdapter(this._context.rpc) : readProvider;` and it would handle other falsy values if applicable

Reply:
It would be `readProvider? readProvider: new RpcReadAdapter(this._context.rpc);` instead